utils/global-state-update-gen/src/generic.rs

```diff
@@ -159,23 +165,69 @@ fn gen_snapshot_from_old(
     mut snapshot: SeigniorageRecipientsSnapshot,
     accounts: &[AccountConfig],
 ) -> SeigniorageRecipientsSnapshot {
-    let stakes_map: BTreeMap<_, _> = accounts
+    // Read the modifications to be applied to the validators set from the config.
+    let validators_map: BTreeMap<_, _> = accounts
         .iter()
-        .filter_map(|acc| acc.stake.map(|stake| (acc.public_key.clone(), stake)))
+        .filter_map(|acc| {
+            acc.validator
+                .as_ref()
+                .map(|validator| (acc.public_key.clone(), validator.clone()))
+        })
         .collect();
 
+    // We will be modifying the entries in the old snapshot passed in as `snapshot` according to
+    // the config.
     for recipients in snapshot.values_mut() {
-        recipients.retain(|public_key, recipient| match stakes_map.get(public_key) {
-            Some(stake) if *stake == U512::zero() => false,
-            Some(stake) => {
-                *recipient =
-                    SeigniorageRecipient::new(*stake, Default::default(), Default::default());
-                true
+        // We use `retain` to drop some entries and modify some of the ones that will be retained.
+        recipients.retain(
+            |public_key, recipient| match validators_map.get(public_key) {
+                // If the validator's stake is configured to be zero, we drop them from the
+                // snapshot.
+                Some(validator) if validator.bonded_amount == U512::zero() => false,
+                // Otherwise, we keep them, but modify the properties.
+                Some(validator) => {
+                    *recipient = SeigniorageRecipient::new(
+                        validator.bonded_amount,
+                        validator
+                            .delegation_rate
+                            // If the delegation rate wasn't specified in the config, keep the one
+                            // from the old snapshot.
+                            .unwrap_or(*recipient.delegation_rate()),
+                        validator
+                            .delegators_map()
+                            // If the delegators weren't specified in the config, keep the ones
+                            // from the old snapshot.
+                            .unwrap_or_else(|| recipient.delegator_stake().clone()),
+                    );
+                    true
+                }
+                // Validators not present in the config will be kept unmodified.
+                None => true,
+            },
+        );
+
+        // Add the validators that weren't present in the old snapshot.
+        for (public_key, validator) in &validators_map {
```

Comment:
Then the network will have more validators than there are slots for the next `auction_delay+1` eras :sweat_smile: But it will be fixed in later eras, as the validators for them will be calculated by the auction contract again.